### PR TITLE
Protect against stackoverflow when printing advice exceptions

### PR DESCRIPTION
**Pull request by JonasKunz**
## What does this PR do?

Tries to mitigate #3152 .

## Checklist

- [x] This is a bugfix
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] ~I have added tests that would fail without this fix~ not feasible to produce a StackOverflow in a test while keeping the system in a known state.

### CODE REVIEW

File: apm-agent-bootstrap/src/main/java/bootstrap/dispatcher/IndyBootstrapDispatcher.java [hunk 1]
```diff
@@ -91,7 +91,11 @@ public static void logAdviceException(Throwable exception) {
                 printStackTrace(exception);
             }
         } catch (Throwable t) {
-            printStackTrace(t);
+            try {
+                printStackTrace(t);
+            } catch (Exception e) {
```
Comment (jackshirazi) [inline apm-agent-bootstrap/src/main/java/bootstrap/dispatcher/IndyBootstrapDispatcher.java:96]:
```suggestion
            } catch (Throwable e) {
```
`StackOverflowError` is an error not exception, so needs to be `Throwable'

Comment (JonasKunz) [reply apm-agent-bootstrap/src/main/java/bootstrap/dispatcher/IndyBootstrapDispatcher.java:96]: Good catch! (pun intended)

**Review by jackshirazi — APPROVED** [review]
Fine without a test but, if you feel like it, you could add a test with a non-tail recursive call that logs on each call until it hits actual stack overflow and confirm the expected "should never happen"
